src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/router/TransitRouter.java

```diff
@@ -70,10 +70,19 @@ public static TransitRouterResult route(
       additionalSearchDays,
       debugTimingAggregator
     );
-    return transitRouter.route();
+    try (
+      var temporaryVertices = new TemporaryVerticesContainer(
```

Comment:
This looks correct but could you quickly explain why this is necessary?

Reply:
I found out that routing crashed to a npe after I added linking with multiple visibility points. The reason was that temporary edges were already disposed at the time when itinerary mapper started collecting  itineraries. Temporary street routing data must be kept alive until the whole routing request is processed.

I am a bit  confused why similar crashes do not happen in current dev-2.x. I saw similar null vertex  references from temporary edges when debugging  dev-2.x branch.